In [1]:
from btgenerate.database.database import Database

In [5]:
db = Database(db="AI_self-driving_workflow")
sdwf = db.pull(table="measured_cond")
metadata = db.pull(table="metadata")
extracted_data = db.pull(table="extracted_data")

In [22]:
sdwf

unique_id  Composition_id   ml_id  Chemical1  Chemical2  Chemical3  \
0            1               0     NaN   0.990000   0.000000   0.000000   
1            2               0     NaN   0.000000   0.990000   0.000000   
2            3               0     NaN   0.000000   0.000000   0.990000   
3            4               0     NaN   0.000000   0.000000   0.000000   
4            5               0     NaN   0.000000   0.000000   0.000000   
..         ...             ...     ...        ...        ...        ...   
512        529               1  1075.0   0.108051   0.108051   0.108051   
513        530               1  1074.0   0.088235   0.088235   0.088235   
514        531               1  1072.0   0.026316   0.000000   0.026316   
515        532               1  1077.0   0.032448   0.044419   0.027511   
516        533               1     NaN   0.083330   0.083330   0.083330   

     Chemical4  Chemical5  Chemical6  Chemical7  Chemical8  Chemical9  \
0     0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   
1     0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   
2     0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   
3     0.990000   0.000000   0.000000   0.000000   0.000000   0.000000   
4     0.000000   0.990000   0.000000   0.000000   0.000000   0.000000   
..         ...        ...        ...        ...        ...        ...   
512   0.112288   0.108051   0.108051   0.112288   0.112288   0.108051   
513   0.029412   0.088235   0.088235   0.147059   0.088235   0.088235   
514   0.131579   0.149123   0.131579   0.026316   0.149123   0.149123   
515   0.024886   0.099190   0.074173   0.241620   0.020536   0.017704   
516   0.083330   0.083330   0.083330   0.083330   0.083330   0.083330   

     Chemical10  Chemical11  Chemical12  Chemical13  Chemical14  Chemical15  \
0      0.000000    0.000000    0.000000         0.0         0.0         0.0   
1      0.000000    0.000000    0.000000         0.0         0.0         0.0   
2      0.000000    0.000000    0.000000         0.0         0.0         0.0   
3      0.000000    0.000000    0.000000         0.0         0.0         0.0   
4      0.000000    0.000000    0.000000         0.0         0.0         0.0   
..          ...         ...         ...         ...         ...         ...   
512    0.000000    0.000000    0.000000         0.0         0.0         0.0   
513    0.029412    0.029412    0.147059         0.0         0.0         0.0   
514    0.026316    0.026316    0.149123         0.0         0.0         0.0   
515    0.173644    0.219078    0.024791         0.0         0.0         0.0   
516    0.083330    0.083330    0.083330         0.0         0.0         0.0   

     Chemical16  measured_conductivity  
0           0.0                0.00238  
1           0.0                0.00630  
2           0.0                0.00570  
3           0.0                0.00613  
4           0.0                0.00127  
..          ...                    ...  
512         0.0                0.00831  
513         0.0                0.00845  
514         0.0                0.00730  
515         0.0                0.00610  
516         0.0                0.00781  

[517 rows x 20 columns]

In [21]:
extracted_data

LiPF6     LiFSI        L1     LiBOB     LiBF4        EC       EMC  \
0    0.038498  0.049998  0.012579  0.005873  0.024701  0.350825  0.179402   
1    0.055435  0.018166  0.028141  0.000948  0.034877  0.406837  0.145816   
2    0.027189  0.062640  0.022794  0.005230  0.009678  0.379067  0.195576   
3    0.011938  0.095931  0.023848  0.005781  0.012846  0.367215  0.161333   
4    0.004931  0.077205  0.012624  0.003491  0.030175  0.367260  0.227624   
..        ...       ...       ...       ...       ...       ...       ...   
379  0.049919  0.066102  0.016666  0.007678  0.016218  0.414293  0.234711   
380  0.040765  0.070588  0.013320  0.005344  0.017548  0.337983  0.160989   
381  0.043737  0.007895  0.014259  0.010249  0.026234  0.356708  0.151480   
382  0.027508  0.085812  0.003569  0.001355  0.040296  0.298796  0.096382   
383  0.038498  0.049998  0.012579  0.005873  0.024701  0.350825  0.179402   

          DMC  measured_conductivity  
0    0.171423                0.00744  
1    0.261021                0.00729  
2    0.183491                0.00815  
3    0.205882                0.00878  
4    0.139636                0.00644  
..        ...                    ...  
379  0.179582                0.00831  
380  0.176994                0.00845  
381  0.205228                0.00730  
382  0.202414                0.00610  
383  0.171423                0.00781  

[384 rows x 9 columns]

In [20]:
print(metadata.loc[1].values)
print(metadata.loc[1, "note"])

[1
 'EC:EMC 1:1 vol; EC/DMC 1:1 vol; Li Salt:Solvent 3:7 wt*\n* Except:\nLiPF6:EC/EMC 16.2:83.8 wt\nL1:EC/EMC 6.72:93.28 wt\nLiBOB:EC/EMC 1.488:98.512 wt\nLiBF4:EC/EMC 15.01:84.99 wt\nL1:EC/DMC 8.376:91.624 wt\nLiBOB:EC/DMC 5.56:94.44 wt\nLiBF4:EC/DMC 14.632:85.368 wt'
 'LiPF6_EC/EMC' 'LiFSI_EC/EMC' 'L1_EC/EMC' 'LiBOB_EC/EMC' 'LiBF4_EC/EMC'
 'LiPF6_EC/DMC' 'LiFSI_EC/DMC' 'L1_EC/DMC' 'LiBOB_EC/DMC' 'LiBF4_EC/DMC'
 'EMC' 'DMC' None None None None]
EC:EMC 1:1 vol; EC/DMC 1:1 vol; Li Salt:Solvent 3:7 wt*
* Except:
LiPF6:EC/EMC 16.2:83.8 wt
L1:EC/EMC 6.72:93.28 wt
LiBOB:EC/EMC 1.488:98.512 wt
LiBF4:EC/EMC 15.01:84.99 wt
L1:EC/DMC 8.376:91.624 wt
LiBOB:EC/DMC 5.56:94.44 wt
LiBF4:EC/DMC 14.632:85.368 wt


In [4]:
df1.loc[df1["Composition_id"]==1]

unique_id  Composition_id   ml_id  Chemical1  Chemical2  Chemical3  \
133        134               1     NaN   0.083330   0.083330   0.083330   
134        135               1     NaN   0.188783   0.017968   0.084670   
135        136               1     NaN   0.030560   0.036504   0.213047   
136        137               1     NaN   0.046547   0.231656   0.102858   
137        138               1     NaN   0.021019   0.002126   0.182634   
..         ...             ...     ...        ...        ...        ...   
512        529               1  1075.0   0.108051   0.108051   0.108051   
513        530               1  1074.0   0.088235   0.088235   0.088235   
514        531               1  1072.0   0.026316   0.000000   0.026316   
515        532               1  1077.0   0.032448   0.044419   0.027511   
516        533               1     NaN   0.083330   0.083330   0.083330   

     Chemical4  Chemical5  Chemical6  Chemical7  Chemical8  Chemical9  \
133   0.083330   0.083330   0.083330   0.083330   0.083330   0.083330   
134   0.016020   0.030702   0.082839   0.042585   0.268044   0.012766   
135   0.133815   0.011101   0.074127   0.172297   0.101208   0.058249   
136   0.010178   0.018271   0.014660   0.088113   0.202197   0.101243   
137   0.172805   0.112426   0.005086   0.255224   0.004188   0.016543   
..         ...        ...        ...        ...        ...        ...   
512   0.112288   0.108051   0.108051   0.112288   0.112288   0.108051   
513   0.029412   0.088235   0.088235   0.147059   0.088235   0.088235   
514   0.131579   0.149123   0.131579   0.026316   0.149123   0.149123   
515   0.024886   0.099190   0.074173   0.241620   0.020536   0.017704   
516   0.083330   0.083330   0.083330   0.083330   0.083330   0.083330   

     Chemical10  Chemical11  Chemical12  Chemical13  Chemical14  Chemical15  \
133    0.083330    0.083330    0.083330         0.0         0.0         0.0   
134    0.206864    0.015607    0.033153         0.0         0.0         0.0   
135    0.054756    0.050473    0.063862         0.0         0.0         0.0   
136    0.069051    0.003381    0.111846         0.0         0.0         0.0   
137    0.090894    0.003686    0.133368         0.0         0.0         0.0   
..          ...         ...         ...         ...         ...         ...   
512    0.000000    0.000000    0.000000         0.0         0.0         0.0   
513    0.029412    0.029412    0.147059         0.0         0.0         0.0   
514    0.026316    0.026316    0.149123         0.0         0.0         0.0   
515    0.173644    0.219078    0.024791         0.0         0.0         0.0   
516    0.083330    0.083330    0.083330         0.0         0.0         0.0   

     Chemical16  measured_conductivity  
133         0.0                0.00744  
134         0.0                0.00729  
135         0.0                0.00815  
136         0.0                0.00878  
137         0.0                0.00644  
..          ...                    ...  
512         0.0                0.00831  
513         0.0                0.00845  
514         0.0                0.00730  
515         0.0                0.00610  
516         0.0                0.00781  

[384 rows x 20 columns]

EC:EMC 1:1 vol; EC/DMC 1:1 vol; Li Salt:Solvent 3:7 wt*
* Except:
LiPF6:EC/EMC 16.2:83.8 wt
L1:EC/EMC 6.72:93.28 wt
LiBOB:EC/EMC 1.488:98.512 wt
LiBF4:EC/EMC 15.01:84.99 wt
L1:EC/DMC 8.376:91.624 wt
LiBOB:EC/DMC 5.56:94.44 wt
LiBF4:EC/DMC 14.632:85.368 wt